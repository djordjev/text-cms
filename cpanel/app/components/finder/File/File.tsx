import { Link } from "@remix-run/react";
import React from "react";

export interface FileProps {
  name: string;
}

export const File: React.FC<FileProps> = (props) => {
  const { name } = props;

  return (
    <Link className="u-flex u-items-center u-flex-col u-group" to={name}>
      <img
        alt={name}
        height={60}
        loading="lazy"
        src="/images/file.png"
        width={60}
      />
      <span className="group-hover:u-underline">{name}</span>
    </Link>
  );
};
