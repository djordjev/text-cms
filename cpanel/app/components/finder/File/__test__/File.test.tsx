import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import { File, FileProps } from '../File';

test('ContextMenu', () => {
  const props: FileProps = { id: 42, name: 'file name' };

  const createComponent = () => {
    const RemixStub = createRemixStub([
      { path: '/', Component: () => <File {...props} /> }
    ]);

    return <RemixStub />;
  };

  const { container } = render(createComponent());

  expect(screen.getByText(props.name)).toBeInTheDocument();

  expect(container.childNodes).toMatchSnapshot();
});
