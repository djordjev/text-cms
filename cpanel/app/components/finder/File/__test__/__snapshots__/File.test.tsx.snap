// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContextMenu 1`] = `
NodeList [
  <a
    class="u-flex u-items-center u-flex-col u-group"
    href="/editor/42"
  >
    <img
      alt="file name"
      height="60"
      loading="lazy"
      src="/images/file.png"
      width="60"
    />
    <span
      class="group-hover:u-underline u-max-w-full u-text-ellipsis u-overflow-hidden u-text-nowrap"
    >
      file name
    </span>
  </a>,
]
`;
